lib/topological_inventory/core/service_catalog_client.rb

```diff
@@ -0,0 +1,57 @@
+require "rest_client"
+require "topological_inventory/core/service_plan_client"
+
+module TopologicalInventory
+  module Core
+    class ServiceCatalogClient
+      def initialize(source)
+        @source = source
+      end
+
+      # TODO? Move this to a module in the API repo
+      def order_service_plan(plan_name, service_offering_name, additional_parameters)
+        payload = ServicePlanClient.new.build_payload(
+          plan_name, service_offering_name, catalog_parameters(additional_parameters)
+        )
+        response = make_request(:post, order_service_plan_url, payload)
+        JSON.parse(response.body)
+      end
+
+      private
+
+      def order_service_plan_url
+        URI.join(@source.default_endpoint.base_url_path, "apis/servicecatalog.k8s.io/v1beta1/namespaces/default/serviceinstances").to_s
+      end
+
+      def make_request(method, url, payload, headers = generic_headers)
+        request_options = {
+          :method     => method,
+          :url        => url,
+          :headers    => headers,
+          :verify_ssl => verify_ssl_mode,
+          :payload    => payload
+        }
+
+        RestClient::Request.new(request_options).execute
+      end
+
+      def generic_headers
+        {
+          "Authorization" => "Bearer #{@source.default_endpoint.authentications.first.password}",
+          "Content-Type"  => "application/json",
+          "Accept"        => "application/json"
+        }
+      end
+
+      def verify_ssl_mode
+        @source.default_endpoint.verify_ssl ? OpenSSL::SSL::VERIFY_PEER : OpenSSL::SSL::VERIFY_NONE
+      end
+
+      def catalog_parameters(parameters)
```

Comment:
@agrare 
I think this might be the problem, it needs to be a simple hash, in our earlier design we were using an array with name, value, type as separate keys and this was converting it to a simple hash. You should be able to directly use the parameters coming in as long as its a simple hash which looks like this



```
params = {
    "NAME" => "madhu-nginx-example-4000",
    "NAMESPACE" => "openshift",  
    "NGINX_VERSION" => "1.12", 
    "MEMORY_LIMIT" => "512Mi", 
    "SOURCE_REPOSITORY_URL" => "https://github.com/sclorg/nginx-ex.git"
  }

  post_params = {
    "apiVersion" => "servicecatalog.k8s.io/v1beta1",
    "kind"=> "ServiceInstance",
    "metadata" => {
      "name" => service_instance_name,
      "namespace" => namespace
     },
    "spec" => {
       "clusterServiceClassExternalName" => svc_class_name,
       "clusterServicePlanExternalName"  => plan_name,
       "parameters"                      => params
     }
  }
```